src/pkg/control/restore.go

```diff
@@ -90,3 +94,62 @@ func EnsureRestoreConfigDefaults(
 
 	return rc
 }
+
+// ---------------------------------------------------------------------------
+// pii control
+// ---------------------------------------------------------------------------
+
+var (
+	// interface compliance required for handling PII
+	_ clues.Concealer = &RestoreConfig{}
+	_ fmt.Stringer    = &RestoreConfig{}
+
+	// interface compliance for the observe package to display
+	// values without concealing PII.
+	_ clues.PlainStringer = &RestoreConfig{}
+)
+
+func (rc RestoreConfig) marshal() string {
+	bs, err := json.Marshal(rc)
+	if err != nil {
+		return "err marshalling"
+	}
+
+	return string(bs)
+}
+
+func (rc RestoreConfig) concealed() RestoreConfig {
+	return RestoreConfig{
+		OnCollision:       rc.OnCollision,
+		ProtectedResource: clues.Hide(rc.ProtectedResource).Conceal(),
+		Location:          path.LoggableDir(rc.Location),
+		Drive:             clues.Hide(rc.Drive).Conceal(),
+	}
+}
+
+// Conceal produces a concealed representation of the config, suitable for
+// logging, storing in errors, and other output.
+func (rc RestoreConfig) Conceal() string {
+	return rc.concealed().marshal()
+}
+
+// Format produces a concealed representation of the config, even when
+// used within a PrintF, suitable for logging, storing in errors,
+// and other output.
+func (rc RestoreConfig) Format(fs fmt.State, _ rune) {
+	fmt.Fprint(fs, rc.concealed())
+}
+
+// String returns a string that contains all path config joined together.
+// Configs that need escaping are escaped.  The result is not concealed, and
```

Comment:
this comment needs some revisions as it doesn't quite make sense

Reply:
done